atomate/vasp/fireworks/core.py

```diff
@@ -325,7 +345,18 @@ def __init__(
             structure.composition.reduced_formula if structure else "unknown", name
         )
 
-        if prev_calc_dir:
+        if spec_structure_key is not None:
+            vasp_input_set = vasp_input_set or MPStaticSet(
+                structure, **vasp_input_set_params
+            )
+            t.append(
+                WriteVaspFromIOSet(
+                    structure=structure,
+                    vasp_input_set=vasp_input_set,
+                    spec_structure_key="prev_calc_structure",
+                )
+            )
+        elif prev_calc_dir:
```

Comment:
This allows the spec to be used to pass the structure.  All the other changes in this file are formatting.